compiler/frontend/src/org/jetbrains/kotlin/psi/synthetics/SyntheticClassOrObjectDescriptor.kt

```diff
@@ -50,13 +50,15 @@ class SyntheticClassOrObjectDescriptor(
     val syntheticDeclaration: KtPureClassOrObject = SyntheticDeclaration(parentClassOrObject, name.asString())
 
     private val thisDescriptor: SyntheticClassOrObjectDescriptor get() = this // code readability
+    var typeParameters: List<TypeParameterDescriptor> = emptyList()
```

Comment:
TBH I'd prefer this to be `private lateinit var` + an `init` method that takes type parameters as a parameter, and require everyone to call `init` after construction explicitly, for clarity. What do you think?

Reply:
In most cases, `emptyList()` is a reasonable default and allows to use existent code without modifications. Personally, I think that such `init` methods are error-prone because it's easy to forget them. 
However, I can see that a lot of descriptor entities in compiler have `initialize` method so I think it's a preferred code style and your comment is fair.